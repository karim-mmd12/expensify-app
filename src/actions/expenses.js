import { v4 as uuid } from 'uuid';
// ADD_EXPENSE
// Takes an object and generate an action from it
export const addExpense = (
    {
        description = '',
        note = '',
        amount = '',
        createdAt = 0
    } = {}
) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
});

// REMOVE_EXPENSE
export const removeExpense = (
    {
        id
    } = {}
) => ({
    type: 'REMOVE_EXPENSE',
    id
});

// Edit_EXPENSE
export const editExpense = (
    id,
    updates
) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});