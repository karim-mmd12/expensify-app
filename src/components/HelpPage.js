import React from 'react';

const HelpPage = () => (
    <div>
        This is from HelpPage Component.
    </div>
);
export default HelpPage;