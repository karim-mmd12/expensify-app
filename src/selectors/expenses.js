import moment from 'moment';

/* Get visible expenses filtered by date and text */
export default (expenses, { text, sortBy, startDate, endDate }) => {
    console.log('expenses', expenses);
    return expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt);
        const startDateMatch = startDate ? createdAtMoment.isSameOrAfter(startDate, 'day') : true;
        const endDateMatch = endDate ? createdAtMoment.isSameOrBefore(endDate, 'day') : true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        /* console.log(`
        ${startDateMatch}
        ${startDateMatch}
        ${startDateMatch}`); */
        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return moment(a.createdAt).isBefore(moment(b.createdAt), 'day') ? 1 : -1;
        } else if (sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
        }
    });
}