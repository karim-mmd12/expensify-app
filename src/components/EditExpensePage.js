import React from 'react'
import ExpenseForm from './ExpenseForm'
import { connect } from 'react-redux'
import { editExpense, startRemoveExpense } from '../actions/expenses'

const EditExpensePage = (props) => {
  console.log(props)
  return (
    <div>
      <ExpenseForm
        expense={props.expense}
        onSubmit={(expense) => {
          props.dispatch(editExpense(props.expense.id, expense))
          props.history.push('/')
        }}
      />

      <button
        onClick={() => {
          console.log(props.expense)
          props.dispatch(startRemoveExpense({ id: props.expense.id }))
          props.history.push('/')
        }}
      >
        Remove
      </button>
    </div>
  )
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id),
  }
}
export default connect(mapStateToProps)(EditExpensePage)
