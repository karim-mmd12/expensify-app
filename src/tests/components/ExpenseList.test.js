import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixures/expenses';
import { ExpenseList } from '../../components/ExpenseList';

test('should return the list of expenses', () => {
    expect(shallow(<ExpenseList expenses={expenses} />)).toMatchSnapshot();
});

test('should return an empty list of expenses with empty message', () => {
    expect(shallow(<ExpenseList expenses={[]} />)).toMatchSnapshot();
});