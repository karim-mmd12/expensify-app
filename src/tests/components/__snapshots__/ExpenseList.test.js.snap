// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return an empty list of expenses with empty message 1`] = `
<div>
  <p>
    No expenses yet!
  </p>
</div>
`;

exports[`should return the list of expenses 1`] = `
<div>
  <ExpenseListItem
    amount={4500}
    createdAt={0}
    description="Rent"
    id="84c57dae-13af-465a-bbfe-3f4ab8e9a579"
    key="84c57dae-13af-465a-bbfe-3f4ab8e9a579"
    note=""
  />
  <ExpenseListItem
    amount={4300}
    createdAt={-172800000}
    description="laudary"
    id="84c57dae-13af-465a-bbfe-3f4ab8e9a571"
    key="84c57dae-13af-465a-bbfe-3f4ab8e9a571"
    note=""
  />
  <ExpenseListItem
    amount={1000}
    createdAt={86400000}
    description="car"
    id="84c57dae-13af-465a-bbfe-3f4ab8e9a572"
    key="84c57dae-13af-465a-bbfe-3f4ab8e9a572"
    note=""
  />
</div>
`;
